site/src/pages/tutorials/tutorial-blog-rest-step3.mdx

```diff
@@ -0,0 +1,223 @@
+
+```java filename=BlogService.java highlight=2
+@Post("/blogs")
+@RequestConverter(BlogPostRequestConverter.class)
+public void createBlogPost(BlogPost blogPost)
+    throws JsonProcessingException {
+  // Implement blog service
+}
+```
+
+## 3. Implement service code
+
+When the request for creation is received, our request converter creates an instance of a blog post object for us. We want to save the blog post object in the map (`blogPosts`) created in the `BlogService` class.
+
+Let's store the blog post information in the map by adding line 5, in the `createBlogPost()` method.
+
+```java filename=BlogService.java highlight=5 showlineno=true
+@Post("/blogs")
+@RequestConverter(BlogPostRequestConverter.class)
+public void createBlogPost(BlogPost blogPost)
+    throws JsonProcessingException {
+  blogPosts.put(blogPost.getId(), blogPost);
+}
+```
+
+## 4. Return response
+
+Now, it's time to return a response to our client. As the response, on top of the information received, add the ID of the post, created time, plus the modified time which would be identical to the created time.
+
+Let's return a response for blog post creation:
+
+1. Replace the return type of the `createBlogPost()` method from `void` to `HttpResponse`.
+2. Create and return an HTTP response using Armeria's <type://HttpResponse> with the information of the post created.
+
+  ```java filename=BlogService.java highlight=4,7
+  import com.linecorp.armeria.common.HttpResponse;
+  import com.linecorp.armeria.common.HttpStatus;
+
+  public HttpResponse createBlogPost(BlogPost blogPost)
+      throws JsonProcessingException {
+    ...
+    return HttpResponse.ofJson(blogPost);
+  }
+  ```
+
+## 5. Test creating a blog post
+
+1. Add our blog service to the server, by removing the dummy service we added in [Step 1. Create a server](/tutorials/tutorial-blog-rest-step1#1-create-a-server-instance) and adding line 6.
+
+  ```java filename="Main.java" highlight=6 showlineno=true
+  public final class Main {
+    ...
+    static Server newServer(int port) {
+      ServerBuilder sb = Server.builder();
+      return sb.http(port)
+               .annotatedService(new BlogService())
+               .build();
+  }
+  ...
+  ```
+
+2. Run the server like we did in [Step 1. Create a server](/tutorials/tutorial-blog-rest-step1) by running the `main()` method or using Gradle.
+When you see the message, "Server has been started", you can try testing service methods.
+3. Call the service method for creating a blog post. Here, we'll use the cURL command.
+
+  ```bash
+  $ curl --location --request POST 'localhost:8080/blogs' \
+  -H 'Content-Type: application/json' \
+  -d '{"title":"My first blog", "content":"Hello Armeria!"}'
+  ```
+
+4. Check the return value.
+
+  ```bash
+  {"id":0,"title":"My first blog","content":"Hello Armeria!","createdAt":1624489646371,"modifiedAt":1624489646371}
```

Comment:
nit: let's just use `...` for all timestamps. 😉 
```suggestion
  {"id":0,"title":"My first blog","content":"Hello Armeria!","createdAt":...,"modifiedAt":...}
```

Reply:
Done ab83cf2e802577fed3da3f8085929d464bbeb313